### PR TITLE
Make UE IP pools configuration optional

### PR DESCRIPTION
**Pull request by daniele-moro**
UE IP pools configuration is now optional, meaning that the UE IP pool interface entries can be pushed directly via PFCPiface without the need of passing that configuration via UP4 netcfg.

### CODE REVIEW

**Comment by daniele-moro** [comment]
@osinstom @pierventre

File: app/app/src/main/java/org/omecproject/up4/config/Up4Config.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
     public static final String MAX_UES = "maxUes";
     public static final String DEVICE_ID = "deviceId"; // TODO: remove this field after all configs updated
     public static final String DEVICES = "devices";
-    public static final String UE_POOLS = "uePools";
+    public static final String UE_POOLS = "uePools"; // Optional, if not provided PFCPiface is free to install UE pools.
```
Comment (ccascone) [inline app/app/src/main/java/org/omecproject/up4/config/Up4Config.java:32]:
```suggestion
    public static final String UE_POOLS = "uePools"; // Optional, if not provided pfcpiface is free to install the corresponding interface entry.
```

File: app/app/src/main/java/org/omecproject/up4/impl/Up4AdminService.java [hunk 2]
```diff
@@ -45,6 +46,14 @@
      */
     void installUpfEntities();
 
+    /**
+     * Gets the UPF interfaces present in the UP4 config file.
+     * This method doesn't query the UPF data plane.
+     *
+     * @return a collection of UPF interfaces configured via UP4 config file.
+     */
+    Collection<UpfInterface> configFileInterfaces();
```
Comment (ccascone) [inline app/app/src/main/java/org/omecproject/up4/impl/Up4AdminService.java:55]:
```suggestion
    /**
     * Gets the UPF interfaces present in the UP4 netcfg.
     * This method doesn't query the UPF data plane.
     *
     * @return a collection of UPF interfaces configured via UP4 netcfg.
     */
    Collection<UpfInterface> configInterfaces();
```

**Review by ccascone — APPROVED** [review]
LGTM. Left only minor comments. Did you want to modify/add the STC tests in this PR or another one?

**Comment by daniele-moro** [comment]
> LGTM. Left only minor comments. Did you want to modify/add the STC tests in this PR or another one?

Now STCs push the UE pool via p4rt

**Comment by osinstom** [comment]
@daniele-moro should we also add the following configuration to `config/pfcp-agent/config.json` ?

```
  "cpiface": {
    "ue_ip_pool": "17.0.0.0/16"
  },
```

**Comment by daniele-moro** [comment]
> @daniele-moro should we also add the following configuration to `config/pfcp-agent/config.json` ?
> 
> ```
>   "cpiface": {
>     "ue_ip_pool": "17.0.0.0/16"
>   },
> ```
> 
> This is required for STC to pass with [omec-project/upf-epc#461](https://github.com/omec-project/upf-epc/pull/461)

Sure, I haven't done that, because I'm doing p4rt based UE IP pool also in pfcp-based STCs. I left a comment into `pfcp-smoke.xml` to remove that when PFCP supports it.

**Review by ccascone — APPROVED** [review]

File: .env.stable [hunk 3]
```diff
@@ -2,7 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-PFCP_AGENT_IMAGE=omecproject/upf-epc-pfcpiface:master-675c303
+PFCP_AGENT_IMAGE=ghcr.io/omec-project/upf-epc/upf-epc-pfcpiface:PR-461
```
Comment (daniele-moro) [inline .env.stable:5]: We should update this before merging, to the right commit SHA after https://github.com/omec-project/upf-epc/pull/461 will be merged.

Comment (daniele-moro) [reply .env.stable:5]: @osinstom

**Pull request merged**